steps/terraform-precheck.yaml

```diff
@@ -0,0 +1,65 @@
+parameters:
+  - name: adolibrary
+    default: cnp-azuredevops-libraries
+
+  - name: environment
+    default: sandbox
+
+  - name: keyvaultName
+    default: ''
+
+  - name: keyvaultSecret
+    default: ''
+
+  - name: location
+    default: 'UK South'
+    values:
+      - 'UK South'
+      - 'UK West'
+
+  - name: serviceConnection
+    default: ''
+
+  - name: tfversion
+    default: ''
+
+steps:
+  - checkout: self
+  - checkout: ${{ parameters.adoLibrary }}
+  - bash: echo ${{ parameters.environment }}
+
+  - task: TerraformInstaller@0
+    displayName: Terraform install
+    inputs:
+      terraformVersion: ${{ parameters.tfversion }}
+
+  - task: AzureKeyVault@1
+    displayName: Retrieve keyvault secret for ADO token
+    inputs:
+      ConnectedServiceName: ${{ parameters.serviceConnection }}
+      keyVaultName: ${{ parameters.keyvaultName }}
+      secretsFilter: ${{ parameters.keyvaultSecret }}
+      runAsPreJob: false
+
+  - task: Bash@3
+    displayName: Prevent parallel run
+    env:
+      thisbuild: $(Build.BuildId)
+      pipelinedefinition: $(System.DefinitionId)
+      azuredevopstoken: $(azure-devops-token)
+    inputs:
+      filePath: $(System.DefaultWorkingDirectory)/${{ parameters.adoLibrary }}/scripts/builds-check.sh
+
+  - task: PowerShell@2
```

Comment:
tests should run before parallel run check, shouldn't need to wait ~5-20 mins to have unit tests fail

Reply:
agreed and moved.